hal/src/argon/platform_ncp_argon.cpp

```diff
@@ -117,12 +117,33 @@ int platform_ncp_fetch_module_info(hal_system_info_t* sys_info, bool create) {
                 info->platform_id = PLATFORM_ID;
                 info->module_function = MODULE_FUNCTION_NCP_FIRMWARE;
 
-                module->info = info;
                 // assume all checks pass since it was validated when being flashed to the NCP
                 module->validity_result = module->validity_checked;
+
+                // IMPORTANT: a valid suffix with SHA is required for the communication layer to detect a change
+                // in the SYSTEM DESCRIBE state and send a HELLO after the NCP update to
+                // cause the DS to request new DESCRIBE info
+                auto suffix = new module_info_suffix_t();
+                if (!suffix) {
+                    delete info;
+                    return SYSTEM_ERROR_NO_MEMORY;
+                }
+                memset(suffix, 0, sizeof(module_info_suffix_t));
+
+                // FIXME: NCP firmware should return some kind of a unique string/hash
+                // For now we simply fill the SHA field with version
+                for (uint16_t* sha = (uint16_t*)suffix->sha;
```

Comment:
any reason why we don't append the module suffix to the NCP firmware as part of the build?

Reply:
The NCP firmware resides in the NCP flash memory which we don't have access directly to. The FIXME here still stands: we need some kind of an AT command that would return it for us.